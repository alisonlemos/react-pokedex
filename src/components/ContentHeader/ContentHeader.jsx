import SelectOrder from './components/SelectOrder';
import TextFilter from './components/TextFilter';
import './ContentHeader.scss';

function ContentHeader() {
	return (
		<section className="content__header">
			<TextFilter />
			<SelectOrder />
		</section>
	);
}

export default ContentHeader;
