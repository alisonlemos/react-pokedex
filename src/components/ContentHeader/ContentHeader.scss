@import '@/assets/colors.scss';

.content__header {
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin: 1rem 0;
}

@media (max-width: 800px) {
	.content {
		&__header {
			align-items: flex-start;
			flex-direction: column;
			gap: 1rem;
		}

		&__actions {
			display: flex;
			gap: 1rem;
			justify-content: space-between;
			width: 100%;
		}
	}
}

@media (max-width: 425px) {
	.content {
		&__actions {
			flex-direction: column;
		}
	}
}