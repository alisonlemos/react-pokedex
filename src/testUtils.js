import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { orderValues } from './services';
import { setupStore } from './store';

export const mockedPokemonList = {
	data: {
		results: [
			{
				national_number: '99999',
				sprites: {
					normal: 'normal_sprite_url',
				},
				name: 'Pokemon Name',
				type: ['Type_1', 'Type_2'],
			},
		],
	},
};

export const initialState = {
	pokemonList: [],
	favoriteList: [],
	categoriesList: [],
	searchKey: '',
	order: orderValues.NATIONAL_NUMBER_ASC,
	favoriteFilter: false,
	categoriesFilter: [],
	loading: false,
};

export function renderWithProviders(
	ui,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}

	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
