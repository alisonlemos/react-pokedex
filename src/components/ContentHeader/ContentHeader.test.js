import { renderWithProviders } from '@/testUtils';
import ContentHeader from './ContentHeader';

describe('Componente ContentHeader', () => {
	it('deve renderizar o componente de acordo com o snapshot', () => {
		expect(renderWithProviders(<ContentHeader />)).toMatchSnapshot();
	});
});
