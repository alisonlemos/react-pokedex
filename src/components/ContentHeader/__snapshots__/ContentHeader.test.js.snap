// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente ContentHeader deve renderizar o componente de acordo com o snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="content__header"
      >
        <div
          class="MuiFormControl-root textFilter__container css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd textFilter__input css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              placeholder="Pesquise por nome ou número"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="filtro de busca por texto"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium textFilter__icon css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root pokedex__selectOrder css-1nrlq1o-MuiFormControl-root"
        >
          <label
            for="select-order"
          >
            Ordenar por 
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select-order"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="select-order"
              role="button"
              tabindex="0"
            >
              Menor número primeiro
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="NATIONAL_NUMBER_ASC"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="content__header"
    >
      <div
        class="MuiFormControl-root textFilter__container css-1nrlq1o-MuiFormControl-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd textFilter__input css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            placeholder="Pesquise por nome ou número"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="filtro de busca por texto"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium textFilter__icon css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root pokedex__selectOrder css-1nrlq1o-MuiFormControl-root"
      >
        <label
          for="select-order"
        >
          Ordenar por 
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select-order"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select-order"
            role="button"
            tabindex="0"
          >
            Menor número primeiro
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="NATIONAL_NUMBER_ASC"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
